src/Storages/FileLog/DirectoryWatcherBase.cpp

```diff
@@ -142,6 +146,7 @@ void DirectoryWatcherBase::start()
 
 void DirectoryWatcherBase::stop()
 {
+    ::write(event_pipe.fds_rw[1], "\0", 1);
```

Comment:
should we update `stopped` before writing to fd?

Reply:
Indeed, fixed.